py/objarray.c

```diff
@@ -580,6 +644,12 @@ STATIC MP_DEFINE_CONST_DICT(array_locals_dict, array_locals_dict_table);
 STATIC const mp_rom_map_elem_t bytearray_locals_dict_table[] = {
     { MP_ROM_QSTR(MP_QSTR_append), MP_ROM_PTR(&array_append_obj) },
     { MP_ROM_QSTR(MP_QSTR_extend), MP_ROM_PTR(&array_extend_obj) },
+
+    { MP_ROM_QSTR(MP_QSTR_find), MP_ROM_PTR(&buffer_find_obj) },
+    { MP_ROM_QSTR(MP_QSTR_rfind), MP_ROM_PTR(&buffer_rfind_obj) },
+    { MP_ROM_QSTR(MP_QSTR_index), MP_ROM_PTR(&buffer_index_obj) },
+    { MP_ROM_QSTR(MP_QSTR_rindex), MP_ROM_PTR(&buffer_rindex_obj) },
```

Comment:
CPython also supports these on `bytes`, as well as `bytearray`. They are not supported on `array.array`. Worth adding to `bytes` also?

Reply:
I think bytes already has it because it's tied to string.

Reply:
It only half works; it acts too much like `string`. In CPython:
```python
>>> b'123'.find(ord('2'))
1
>>> b'123'.find(b'2')
1
```
But in CircuitPython:
```python
>>> b'123'.find(ord('3'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't convert 'int' object to str implicitly
>>> b'123'.find(b'3')
2
```

Reply:
Ya, I didn't implement the find a single value. I was ok with that since the subset we do support works in CPython. I'm worried about code size a bit.

Reply:
Got it. Yes, I'm wondering about the tiny ports; hope it will fit.